pkg/client/cache/delta_fifo.go

```diff
@@ -132,6 +135,12 @@ var (
 	ErrZeroLengthDeltasObject = errors.New("0 length Deltas object; can't get key")
 )
 
+// Close the queue.
+func (f *DeltaFIFO) Close() {
+	f.closed = true
```

Comment:
Even if there is only one writer and the writer writes only once? In a sense, the ``f.closed`` is false until the ``Close()`` is called. Then it is true forever. The reader can read the ``f.closed`` set to false, then different goroutine calls ``Close()`` which sets the ``f.closed`` to true. Then the reader continues with the false even when it is true now. How does it differ from locking the variable? The critical section is reading the ``f.closed``. Once it is read it does not matter if it is changed right after the reading.